src/components/inputs/CPriceInput.vue

```diff
@@ -0,0 +1,158 @@
+ */
+watch(
+  model,
+  () => {
+    // Update field value to fit format
+    fieldValue.value = model.value
+      ? model.value.toFixed(2).toString().replace(".", ",")
+      : undefined;
+  },
+  { immediate: true },
+);
+
+/**
+ * This component builds an input field for a price with simplified typing experience.
+ *
+ * The format of the price will always be `x,xx`.
+ */
+withDefaults(
+  defineProps<{
+    /**
+     * The label of this field.
+     */
+    label: string;
+
+    /**
+     * The variant of this field.
+     */
+    variant?:
+      | "underlined"
+      | "outlined"
+      | "filled"
+      | "solo"
+      | "solo-inverted"
+      | "solo-filled"
+      | "plain";
+
+    /**
+     * Whether the input fields should be disabled.
+     */
+    disabled?: boolean;
+
+    /**
+     * The rules to use to validate the password.
+     */
+    rules: ((password: string) => true | string)[];
+  }>(),
+  { variant: "outlined" },
+);
+
+/**
+ * Parses the given string value into a price with the format `x,xx`.
+ * The formatted price can easily be parsed into a float value with `stringToFloat`.
+ *
+ * This function is designed to support typing the price from start to end without caring about the comma.
+ *
+ * @param value the value to parse
+ */
+function parsePrice(value?: string): string | undefined {
+  if (!value) return undefined;
+
+  // Remove `,` and `.`
+  value = value.replace(",", "");
+  value = value.replace(".", "");
+
+  // Parse float to remove leading 0s
+  const floatVal: number = parseFloat(value);
+
+  if (isNaN(floatVal) || floatVal === 0) {
+    return undefined;
+  }
+
+  const stringVal = floatVal.toString();
+
+  // Add decimal divider
+  if (stringVal.length === 1) {
+    return "0,0" + stringVal;
+  } else if (stringVal.length === 2) {
+    return "0," + stringVal;
+  } else {
+    return (
+      stringVal.slice(0, -2) +
+      "," +
+      (stringVal.length > 1 ? stringVal.slice(-2) : "0" + stringVal)
+    );
+  }
+}
+
+/**
+ * Parses a string price value into a float.
+ *
+ * @param value the parsed price value as a string.
+ */
+function stringToFloat(value?: string): number | undefined {
+  return value ? parseFloat(value.replace(",", ".")) : undefined;
+}
+
+/**
+ * Parses the input on any change to fit the price format.
+ */
+function onInputChange(): void {
+  const parsed = parsePrice(fieldValue.value);
+  const modelValue = parsed ? stringToFloat(parsed) : undefined;
+
+  if (model.value === modelValue) {
+    // The models value did not change, but the field value probably did.
+    // Example: Changing 12,00 to 12 will both result in a model value of 12, but in a changed fieldValue
+    // If this happens, we need to update the fieldValue here, because the `model` watcher will not be triggered.
+
+    // Update field value to fit format
+    fieldValue.value = model.value
+      ? model.value.toFixed(2).toString().replace(".", ",")
```

Comment:
1. Create a method for this
2. IMO we can move this out of the if statement. I find it kind of confusing that this is done only selectively.

Reply:
it's done selectively to prevent it to be done mutliple times.

If the models value changes, the watcher will update the fieldValue

Reply:
I know, but this is confusing here. I think it would be fine to update it twice. Using the model trigger for that is a bit wonky for that IMO.

Reply:
same as above, I would rather not change it now, this can be improved in the future, but now it works and I think this is better than calling it multiple times

Reply:
okay